### PR TITLE
Fixes and general cleanup across the project 

### PR DESCRIPTION
**Pull request by L1ghtmann**
Changes
---
- Added two more items created when dragon is installed via the dev script to the gitignore
- Updated krit* -> cynder
- Created and used new bash variables for dragon's primary directories:
  - $DRAGON_DIR=.dragon
  - $DRAGON_ROOT_DIR=$HOME/$DRAGON_DIR
  - $DRAGON_DATA_DIR=$PWD/$DRAGON_DIR (when $PWD != $HOME)
- Updated all bash scripts to use double-bracket syntax and write out one liners 
- Fixed a handful of previously-broken dragon options (e.g., remote, checkra1n, send, norm, etc)
- Fixed typos in a handful of comments
- Simplified some of the commands in the bash scripts
- Made common color echo functions to avoid code duplication
- Added missing shebangs and yml headers
- Removed an old DragonPrefs reference 
- Updated test repo links (5/8 fail on Linux due to one reason or another. If you could run the suite to confirm all work, that'd be great!)
- Will now properly determine max job number for ninja
- Updated prerun_checks to have a *complete* dependency check and resolution mech (tested on all platforms but MacOS -- guessed on the deps here please confirm)
- Used new $DRAGON* vars in the wizard
- Cleaned up python file imports
- A handful of smaller things I'm probably forgetting 

Got COVID recently and had some time to give dragon some love. Apologies for all the prs. And thanks again for this awesome project!

### CODE REVIEW

File: bin/dragon [hunk 1]
```diff
@@ -101,11 +110,10 @@ while [ "$1" != "" ]; do
         # --------
         # Commands
         # --------
-
         u | update | upgrade )            source $TOOLPATH/upgrader
                                 drexit
                                 ;;
-        rs | respring )         python3 -m dragon.device run killall -9 SpringBoard
+        rs | respring )         python3 -m dragon.device run sbreload
```
Comment (cxnder) [inline bin/dragon:116]: Revert this line and everything else is good to go.

**Review by cxnder — CHANGES_REQUESTED** [review]

File: bin/dragon (hunk 1 above)
Comment (cxnder) [reply bin/dragon:116]: hmu on discord for an explanation if needed.

**Comment by cxnder** [comment]
Also, thank you sososososo much for this PR, this is amazing 😊 😊 😊 😊

**Comment by L1ghtmann** [comment]
@cxnder Made a handful of minor changes since you last reviewed. Would you mind doing another once over if you're thinking of merging? Thanks!

**Comment by L1ghtmann** [comment]
@MrGcGamer graciously gave this a spin and confirmed it's working as intended. 

If there's nothing else that you'd like changed @cxnder, this should be good to go.

**Comment by cxnder** [comment]
Holy fuck, thanks

**Pull request merged**